File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    const alpha = ['A', 'E', 'I', 'O', 'U'];
    let count = 0;
    let isWord = false;
    
    function dfs(val) {
        if (val.length > 5 || isWord) return;
        
        count++;
        if (val === word) {
            console.log("정답");
            isWord = true;
            return;
        }
        
        for (let i = 0; i < alpha.length; i++) {
            dfs(val + alpha[i]);
        }
    }
    
    for(let i = 0; i < alpha.length; i++) {
        dfs(alpha[i]);
    }
    
    return count;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
alpha = ['A', 'E', 'I', 'O', 'U']
result = []

def dfs(cur):
    if len(cur) > 5:
        return
    result.append(cur)
    
    for a in alpha:
        dfs(cur + a)

def solution(word):
    dfs("")
    return result.index(word)
    